src/leveldbwrapper.h

```diff
@@ -34,7 +34,7 @@ class CLevelDBBatch
 
 public:
     template <typename K, typename V>
-    void Write(const K& key, const V& value)
+    void Write(const K& key, const V& value, const std::string& obfuscate_key = std::string())
```

Comment:
I initially did that, but this is called from contexts where `obfuscate_key` isn't in scope and hasn't been defined, namely `CBlockTreeDB::WriteTxIndex`. Now that I'm looking at it, there're some missing parameterizations of `obfuscate_key` (e.g. [here](https://github.com/jamesob/bitcoin/blob/obfuscate_chainstate/src/txdb.cpp#L36)). 

It isn't immediately obvious to me how to convey `db.obfuscate_key` to that context... might need some refactoring.